components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1045,6 +1063,75 @@ public static String getClientIdForAccessToken(String accessTokenIdentifier) thr
         return accessTokenDO.getConsumerKey();
     }
 
+    /***
+     * Read the configuration file at server start up.
+     * @param tenantId
+     */
+    public static void initTokenExpiryTimesOfSps(int tenantId) {
+        try{
+            Registry registry = OAuth2ServiceComponentHolder.getRegistryService().getConfigSystemRegistry(tenantId);
+            if (!registry.resourceExists(OAuthConstants.TOKEN_EXPIRE_TIME_RESOURCE_PATH)) {
+                Resource resource = registry.newResource();
+                registry.put(OAuthConstants.TOKEN_EXPIRE_TIME_RESOURCE_PATH, resource);
+            }
+        } catch (RegistryException e) {
+            log.error("Error while creating registry collection for :" + OAuthConstants.TOKEN_EXPIRE_TIME_RESOURCE_PATH, e);
+        }
+    }
+
+    /***
+     * Return the SP-token Expiry time configuration object when consumer key is given.
+     * @param consumerKey
+     * @param tenantId
+     * @return A SpOAuth2ExpiryTimeConfiguration Object
+     */
+    public static SpOAuth2ExpiryTimeConfiguration getSpExpiryTimeObject(String consumerKey, int tenantId) {
+        SpOAuth2ExpiryTimeConfiguration spTokenTimeObject = new SpOAuth2ExpiryTimeConfiguration();
+        try {
+            IdentityTenantUtil.initializeRegistry(tenantId, getTenantDomain(tenantId));
+            Registry registry = IdentityTenantUtil.getConfigRegistry(tenantId);
+            if (registry.resourceExists(OAuthConstants.TOKEN_EXPIRE_TIME_RESOURCE_PATH)) {
+                Resource resource = registry.get(OAuthConstants.TOKEN_EXPIRE_TIME_RESOURCE_PATH);
+                String jsonString = "{}";
+                if (((List)resource.getProperties().get(consumerKey)) != null) {
+                    jsonString = ((List) resource.getProperties().get(consumerKey)).get(0).toString();
```

Comment:
possible IOOB exception here. need to refactor the code to efficiently check for null and IOOB.